pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -350,6 +350,36 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
+	if c.spec.Network.MultiClusterService.Enabled {
+		for _, mon := range c.ClusterInfo.Monitors {
+			var failMon bool
+			monResourceName := resourceName(mon.Name)
+			isAlreadyExported, err := k8sutil.IsServiceExported(c.ClusterInfo.Context, c.context, monResourceName, c.ClusterInfo.Namespace)
+			if err != nil {
+				return errors.Wrapf(err, "failed to check if the service %q is already exported", mon.Name)
+			}
+			if !isAlreadyExported {
+				failMon = true
+			} else {
+				exportedIP, err := k8sutil.GetExportedServiceIP(fmt.Sprintf("%s.%s.svc.clusterset.local", monResourceName, c.ClusterInfo.Namespace))
+				if err != nil {
+					return errors.Wrapf(err, "failed to get exported IP of the service %q", mon.Name)
+				}
+				if !strings.Contains(mon.Endpoint, exportedIP) {
+					failMon = true
+				}
+			}
+
+			if failMon {
+				c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, mon.Name)
+				if err != nil {
+					return errors.Wrapf(err, "failed to failover mon %q", mon.Name)
+				}
+				return fmt.Errorf("mon %q pod is running but its service is not exported. Reconcling after mon failover", mon.Name)
```

Comment:
Can't we return success at this point? This doesn't look like an error condition.
```suggestion
				return nil
```

Reply:
right. Returning nil.